src/server/svr_jobfunc.c

```diff
@@ -5906,6 +5906,86 @@ send_job_exec_update_to_mom(job *pjob, char *err_msg, int err_msg_sz,
 	return (rc);
 }
 
+/**
+ *
+ * @brief
+ *	Extracts mom hostnames from exechostx and adds it to list
+ *
+ * @param[in/out]	to_head - destination reliable_job_node list
+ * @param[in]	exechostx - string in exechost format
+ *
+ * @return 	void
+ */
+static void
+populate_mom_list(pbs_list_head *to_head, char *exechostx)
+{
+	char *hostn = NULL, *last = NULL, *peh;
+	int  hasprn = 0;
+
+	if ((to_head == NULL) || (exechostx == NULL) || !*exechostx)
+		return;
+
+	peh = strdup(exechostx);
+	if (peh == NULL) {
+		log_err(errno, __func__, "strdup error");
+		return;
+	}
+
+	CLEAR_HEAD((*to_head));
+
+	for (hostn = parse_plus_spec_r(peh, &last, &hasprn);
+			hostn;
+			hostn = parse_plus_spec_r(last, &last, &hasprn) ) {
+		if (reliable_job_node_add(to_head, strtok(hostn, ":/")) == -1) {
+			free(peh);
```

Comment:
If there's a failure to add a host in reliable_job_node_add(), shouldn't we also free/clear to_head? Otherwise, it's not a complete list.